server/controllers/user.js

```diff
@@ -151,6 +150,83 @@ class User {
       { token },
     );
   }
+
+  /**
+   * This method gets a user profile
+   * @param {object} request
+   * @param {object} response
+   * @returns {object} the user data
+   */
+  static async profile(request, response) {
+    const data = await userRepo.getUserbyUsername(request.params.username);
+    if (data === null) {
+      return badHttpResponse(
+        response,
+        404,
+        'User not found',
+        'No data to show'
+      );
+    }
+    const {
+      id,
+      firstName,
+      lastName,
+      facebook,
+      google,
+      twitter,
+      bio,
+      imageUrl,
+      createdAt,
+      updatedAt,
+    } = data;
+    return goodHttpResponse(
+      response,
+      200,
+      'User found',
+      request.isAuthorized ? data : {
```

Comment:
so if t request.isAuthorized is falsy? Also you shouldn't be doing this check in goodHttpResponse function

Reply:
It returns `{id,
        firstName,
        lastName,
        facebook,
        google,
        twitter,
        bio,
        imageUrl,
        createdAt,
        updatedAt,
      },`
The difference is that `data` has **email** and **isConfirmed**

Reply:
Okay